src/harvesters/_private/frontend/canvas.py

```diff
@@ -223,8 +223,11 @@ def _update_texture(self):
                         update = False
                     else:
                         data_format = buffer.payload.components[0].data_format
-                        effective_data_size = get_effective_data_size(data_format)
-                        exponent = effective_data_size - 8
+                        bpp = get_bits_per_pixel(data_format)
+                        if bpp is not None:
+                            exponent = bpp - 8
+                        else:
+                            update = False
```

Comment:
Not directly related to these changes, but a more general question:
Is all this handling with `update = False` even needed? Meaning do we always have to call `self._draw()`?

I actually would have expected that you only call it if something changed and I would even expect this to increase performance.

Reply:
The Harvester GUI has the "Pause" button and it is used to keep displaying lastly displayed image on the canvas. If we didn't call the `self._draw` method then the OpenGL engine, Vispy, overwrites the canvas with its default backgound data (i.e., the gray image).

Reply:
On the other hand, the `udpate` is used to indicate that we are goint to swap the texture (i.e., the image). The following code is the place where we swap the texture.

    self._program['texture'] = content

Once we called `self._draw` the buffer turns useless then we queue it.

Reply:
Anyway, let me know anytime when your branch is ready to get pulled.

Reply:
Ah, ok... it wasn't clear to me that vispy will overwrite the canvas if we don't call draw (also doesn't really say on http://vispy.org/gloo.html).

I that case it is ready...